server/datastore/mysql/software_installers.go

```diff
@@ -174,8 +174,9 @@ func (ds *Datastore) getOrGenerateSoftwareInstallerTitleID(ctx context.Context,
 	insertArgs := []any{payload.Title, payload.Source}
 
 	if payload.BundleIdentifier != "" {
-		selectStmt = `SELECT id FROM software_titles WHERE bundle_identifier = ?`
-		selectArgs = []any{payload.BundleIdentifier}
+		// match by bundle identifier first, or standard matching if we don't have a bundle identifier match
+		selectStmt = `SELECT id FROM software_titles WHERE bundle_identifier = ? OR (name = ? AND source = ? AND browser = '') ORDER BY bundle_identifier = ? DESC LIMIT 1`
```

Comment:
Curious why the second `ORDER BY bundle_identifier = ? DESC LIMIT 1` is needed (vs simply `ORDER BY bundle_identifier DESC LIMIT 1`)

Reply:
To match existing behavior in as many cases as possible. Basically this will return either zero, one, or two rows:

1. No rows (so new row will be inserted)
2. Only a bundle ID match but no name match (so use that row)
3. No bundle ID match but a match on name (so use that row)
4. Rows exist that both matching bundle ID and matching name, and they aren't the same row (in which case we prefer the row with the bundle ID match)

The order-by ensures that we pull the same row in case 4 as we would have prior to this fix, so we don't trigger a regression for what #19144 was trying to fix.

Reply:
Ah, thanks for the explanation, makes sense now!